lib/fetchContent.js

```diff
@@ -348,6 +348,227 @@ function getArticleRelations(days, facets, aspects)
 		});
 }
 
+function getArticlesAggregation(days, facets, aspects, minCorrelation=2)
+{
+	const date = time.getDatetimeRange('days', days, 0);
+	aspects = checkMetaDataPresent(aspects);
+	const params = {
+		queryString: `lastPublishDateTime:> ${date.next}`,
+		maxResults : 100,
+		aspects : aspects,
+		facets : {
+			"names": facets,
+			"maxElements":-1
+		}
+	};
+
+	return searchDeeper(params, 10)
+		.then(results =>  extractArticles(results) )
+		.then(articles => aggregateArticles(articles, facets, minCorrelation, days, params, filterByMetadataCsv=null) )
+		.then(aggregation => {
+			// look for all the distinct genre CSVs
+			let allCsvs = Object.keys(aggregation.articlesByMetadataCsv);
+			let allGenreCsvs = allCsvs.filter( csv => csv.startsWith('genre:') );
+			// re-establish the full list of articles
+			let articles = Object.values(aggregation.articlesByUuid);
+			// construct an aggregation for each genre csv
+			let aggregationsByGenre = {};
+			allGenreCsvs.forEach( csv => {
+				aggregationsByGenre[csv] = aggregateArticles(articles, facets, minCorrelation, days, params, filterByMetadataCsv=csv);
+			});
+			let articleCounts = {};
+			allGenreCsvs.forEach(csv => {
+				articleCounts[csv] = aggregationsByGenre[csv].numArticles;
+			});
+
+			return {
+				'description' :[
+					'calculating aggregations for each genre CSV',
+				],
+				articleCounts,
+				aggregationsByGenre
+			};
+		})
+		.catch(err => {
+			throw err;
+		});
+}
+
+function addUuidToMetadataTerm(grouping, metadataKey, term, uuid){
```

Comment:
`addUuidToMetadataTerm` and `addFacetCorrelation` seem very similar. Can they be made more generic to avoid repetitions?

Reply:
the code is still evolving, so maybe